tests/process_test_set.py

```diff
@@ -276,6 +277,26 @@ def can_write_to_stdin_of_spawned_process_when_using_pty(shell):
         # Get the output twice since the pty echoes input
         assert_equal(b"hello\r\nhello\r\n", result.output)
 
+    @test
+    def exception_is_raised_if_cwd_does_not_exist(shell):
+        try:
+            shell.run(["echo", "run"], cwd="/some/silly/path")
+            assert_true(False)
+        except spur.RunProcessError as error:
+            # SshShell
+            assert_true(b'No such file or directory' in error.stderr_output)
+        except (IOError, OSError) as e:
+            # LocalShell
+            # FileNotFoundError on Py3.3+ inherits from IOError
+            # OSError is raised on Python 2
+            # errno.ENOENT: No such file or directory
+            assert_equal(e.errno, errno.ENOENT)
```

Comment:
We already wrap up the exception when the command isn't found, so a new exception would at least be internally consistent. Regardless, we should definitely have the same behaviour across `SshShell` and `LocalShell`.